src/app/components/SatButtons/index.tsx

```diff
@@ -5,39 +5,69 @@ import Button from "../Button";
 type Props = {
   onClick: (amount: string) => void;
   disabled?: boolean;
+  min?: number;
+  max?: number;
 };
 
-function SatButtons({ onClick, disabled }: Props) {
+function stringifyThousands(number: number) {
+  return number < 1000
+    ? `${Math.round(number)}`
+    : number < 1_000_000
+    ? `${Math.floor(number / 1000)}K`
+    : `${Math.floor(number / 1_000_000)}M`;
+}
+
+function tensRounder(number: number, index: number) {
+  const rounded = Math.round(number);
+  const tens = Math.round(Math.log(rounded) * Math.LOG10E) - 1;
+  const integralFunction = index === 0 ? Math.ceil : Math.trunc;
+  return integralFunction(rounded / Math.pow(10, tens)) * Math.pow(10, tens);
+}
+
+function amountsArray(min: number, max: number) {
```

Comment:
Looking at this is leaving afraid and confused :P  
Not sure how others feel. Is there a nice way to makes this easier to digest?